### DISCUSSION TITLE
Criteria APIのfrom句内でのサブクエリ

### DISCUSSION THREAD

**Discussion by ironclad-15256304** [domaframework/doma#969] [Q&A] [answered]
### 相談内容

いつもお世話になっております。
私の参加しているプロジェクトでは、下記のようなSQLを必要とするシチュエーションが存在します。

```
select
    t_0_0.*
    t_1.key
    t_1.name
from
    (select *
     from table_a t_0
     where t_0.group_key = 12345
    ) t_0_0
left outer join
    table_b t_1
on
    t_0_0.key = t_1.key
```

メインとなるテーブルは一つですが、大きめのデータと紐付けるたりするため、join→whereではなくfrom句の中で極力事前にデータを絞り込んでおき、その後に他のテーブルと結合を行いたい場合があります。
これを可能であればCriteria APIで行いたいと考えております。

DomaのCriteria API入門記事やサンプル、およびDiscussionを拝見した感じでは、from句内でサブクエリを実行するようなものを見つけられませんでした。
お忙しいところ大変恐縮ですが、上記内容が実現可能かどうかご教示頂けますと幸いです。

### 動作環境

+ Java Framework: Spring boot 3.1.0
+ Doma: 2.53.1（doma-spring-boot 1.7.0）
+ DB: postres もしくは oracle

**Comment by nakamura-to** [comment, marked as answer]
Criteria APIではサポートしていません。
SQLテンプレートをお使いください。

**Reply by ironclad-15256304** [reply]
承知しました。ご回答頂き有難うございます。
